### ISSUE TITLE
Protocol.UndefinedError - enumerating sth in transaction.submit?

### ISSUE THREAD

**Issue by pdobacz** [omisego/elixir-omg#457]
**Message**

```
  protocol Enumerable not implemented for <<59, 2, 183, 56, 142, 245, 12, 130, 103, 40, 127, 169, 160, 207, 222, 39, 152, 66, 125, 147, 170, 16, 225, 175, 119, 219, 2, 115, 185, 134, 34, 228, 47, 8, 137, 157, 41, 115, 214, 191, 239, 141, 75, 45, 144, 79, 102, 227, 227, 188, ...>>. This protocol is implemented for: Ecto.Adapters.SQL.Stream, Postgrex.Stream, DBConnection.Stream, DBConnection.PrepareStream, Stream, Function, HashDict, Map, Date.Range, List, Range, HashSet, MapSet, IO.Stream, File.Stream, GenEvent.Stream
```

**Backtrace (last 10 lines)**

```
(elixir) /root/deb/elixir_1.8.0-1/lib/elixir/lib/enum.ex:1: Enumerable.impl_for!/1
(elixir) /root/deb/elixir_1.8.0-1/lib/elixir/lib/enum.ex:141: Enumerable.reduce/3
(elixir) lib/enum.ex:311: Enum.all?/2
(omg_api) lib/state/transaction/signed.ex:54: OMG.API.State.Transaction.Signed.reconstruct/2
(omg_api) lib/core.ex:48: OMG.API.Core.recover_tx/1
(omg_api) lib/api.ex:31: OMG.API.submit/1
(omg_rpc) lib/web/controllers/transaction.ex:28: OMG.RPC.Web.Controller.Transaction.submit/2
(omg_rpc) lib/web/controllers/transaction.ex:15: OMG.RPC.Web.Controller.Transaction.action/2
(omg_rpc) lib/web/controllers/transaction.ex:15: OMG.RPC.Web.Controller.Transaction.phoenix_controller_pipeline/2
(omg_rpc) lib/web/endpoint.ex:15: OMG.RPC.Web.Endpoint.instrument/4
```

**Comment by pnowosie** [comment]
Looks like badly constructed transaction, instead of list of signatures, single signature's bytes was passed.

**Comment by pdobacz** [comment]
let's make it a `malformed_transaction` case then, I guess? I wonder how it ended up not returning that.

**Comment by pthomalla** [comment]
fix version in master from a7eb218
